library/options.inc.php

```diff
@@ -2021,7 +2079,27 @@ function generate_display_field($frow, $currvalue)
         $s = htmlspecialchars($uname, ENT_NOQUOTES);
     } // billing code
     else if ($data_type == 15) {
-        $s = htmlspecialchars($currvalue, ENT_NOQUOTES);
+        $s = '';
+        if (!empty($currvalue)) {
+          $relcodes = explode(';', $currvalue);
+          foreach ($relcodes as $codestring) {
+            if ($codestring === '') continue;
+            list($codetype, $code) = explode(':', $codestring);
+            $query = "SELECT c.code_text FROM codes AS c, code_types AS ct WHERE " .
+              "ct.ct_key = '$codetype' AND " .
+              "c.code_type = ct.ct_id AND " .
+              "c.code = '$code' AND c.active = 1 " .
+              "ORDER BY c.id LIMIT 1";
```

Comment:
looks like another place where should be using custom/code_types.inc.php function here in order to support external codes